src/taker/offers.rs

```diff
@@ -106,6 +106,18 @@ impl OfferBook {
         self.all_makers.iter().collect()
     }
 
+    /// Gets all good makers that can handle a specific amount.
+    /// Filters makers based on their min_size and max_size limits.
+    pub fn suitable_makers_for_amount(&self, amount: bitcoin::Amount) -> Vec<&OfferAndAddress> {
+        self.all_good_makers()
+            .into_iter()
+            .filter(|oa| {
+                amount >= bitcoin::Amount::from_sat(oa.offer.min_size)
```

Comment:
To handle a edge case of having only the exact amount if min limit, it should be `amount >= min_size + tx_fees + maker's fees`. 

How to get those fees value in this current function scope is the bigger question. 

try to see the easiest way you can either estimate or capture the exact fee values.